server/src/test/java/org/elasticsearch/action/bulk/BulkWithUpdatesIT.java

```diff
@@ -618,5 +621,31 @@ public void testInvalidIndexNamesCorrectOpType() {
         assertThat(bulkResponse.getItems()[1].getOpType(), is(OpType.UPDATE));
         assertThat(bulkResponse.getItems()[2].getOpType(), is(OpType.DELETE));
     }
+
+    public void testNoopUpdate() {
+        String indexName = "test";
+        createIndex(indexName, Settings.builder().put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1).build());
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        ensureGreen(indexName);
+        IndexResponse doc = index(indexName, "_doc", "1", "{\"user\": \"xyz\"}");
```

Comment:
NIT: you could do the document source here and below with the nicer API (I know David recently replaced a bunch of these spots :)):

i.e.:
```java
        IndexResponse doc = index(indexName, "_doc", "1", "user", "xyz");
        assertThat(doc.getShardInfo().getSuccessful(), equalTo(2));
        final BulkResponse bulkResponse = client().prepareBulk()
            .add(new UpdateRequest().index(indexName).id("1").detectNoop(true).doc("user", "xyz"))
```

Reply:
++ I pushed https://github.com/elastic/elasticsearch/pull/46458/commits/9cd3f93f4e1abfcdb8e76f5e1e8e2d83d5ab7a48.